### PR TITLE
Get latest release and verify SHA-256 hash

### LINKED ISSUE #4: Download latest release
Hi! This looks great and I've actually been working on something similar so I'd love to contribute. I noticed this comment:

https://github.com/metalbear-co/sccache-action/blob/91e58973d3f1c9d27d9ae48be6500f687a725fbd/src/main.ts#L9

Using the GitHub API, it's possible to determine the latest version of `sccache` and download the corresponding binary. Would you be ok if I contributed this in a PR?

### LINKED ISSUE #5: Verify SHA-256 of downloaded binary
It may be worth verifying the SHA-256 of the downloaded binary since `sccache` provides them as release assets. Is this something I could contribute in a PR?

### PR DESCRIPTION
**Pull request by rajivshah3**
- Get latest release with GitHub API (closes #4)
  - Fall back to known stable version if latest is not usable
  - Use `${{ secrets.GITHUB_TOKEN }}` to avoid issues with rate limiting
- Verify `sccache` SHA-256 hash (closes #5)

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -58,7 +126,7 @@ async function guardedRun(): Promise<void> {
     return;
```
Comment (rajivshah3) [inline src/main.ts:167]: I'm unable to leave a comment on line 122 (`const sccacheDirectory = find...`), but I'm not sure how to proceed if the `VERSION` constant is removed. If we call `getLatestRelease` to get the version, and the release is published but prebuilt binaries are not available, an error will be thrown. Should we do [`findAllVersions`](https://github.com/actions/toolkit/blob/819157bf872a49cfcc085190da73894e7091c83c/packages/tool-cache/src/tool-cache.ts#L512) instead and choose the latest available version?

Comment (aviramha) [reply src/main.ts:167]: I think we should have a const for "stable"/last known working version for a fallback

Comment (aviramha) [reply src/main.ts:167]: They released 0.3.3, so we can put it now.

**Comment by aviramha** [comment]
Looks good! Is it draft on purpose or I can merge it?

**Comment by rajivshah3** [comment]
@aviramha sorry for the delay in replying! There are two things left imo:

- In my own [tests](https://github.com/rajivshah3/sccache-action/actions/runs/3707175063/jobs/6283312156#step:3:18) I ran into an error that said `Error getting latest release: 403 rate limit exceeded`. I'm going to see if I can reproduce this. I wonder if a `GITHUB_TOKEN` needs to be passed to the action?
  - If we need to use a token, the blue "Important" box [here](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow) makes it sound like we can pick it up without the action user actually passing a token 🤔 
- I don't want to increase the scope of this PR if possible, but I'm wondering how the caching of the `sccache` binary should be handled. Should we always check if the latest version is in the cache and if not, we download it and cache it? Alternatively we could check the cache for any version greater than or equal to the last known working version and use that
  - If we go with the second approach and a bug arises in the version that's in the cache, should the action user have a way of forcing the action to download and use a specific version of `sccache`?
  - A side question — is the tool cache only actually used on self-hosted runners? If that's the case I wonder how we can test this behavior if we're using the GitHub-hosted runners 🤔 

I'll look into the first point but I was wondering what your thoughts are on the second one

**Comment by aviramha** [comment]
Either approach looks good ;)

**Review by aviramha — APPROVED** [review]
Great job!

**Pull request merged**